src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,60 @@
+package duke;
+
+import java.util.ArrayList;
+
+/** Represents a parser that parses input string
+ *    to determine what command is run.
+ */
+public class Parser {
+    private int counter;
+    private ArrayList<Task> tasklist;
```

Comment:
A Parser technically should only parse your command strings and not have access/modify your task list for better encapsulation - consider using Command classes instead